### PR TITLE
Fix the struct buffer issue.

### PR DESCRIPTION
**Pull request by lucyk4t**
It can control the buffer size I guess.

### CODE REVIEW

**Comment by pietermarsman** [comment]
Which issue is this pull-request solving?

**Comment by pietermarsman** [comment]
Hi @lucyk4t, your description is a little short. And I cannot find any issues that are related to this `.unpack()` statement. Could you describe the problem in more detail (here or in an issue) and how you've solved the problem?

**Comment by lucyk4t** [comment]
when i tried to use it for analyzing pdf and then error code is printed.
so that I tried to modify in your code and then that error is not founded. sorry for late answer. 

Traceback (most recent call last):
  File "D:/Pycharm/ValTest/parsetest.py", line 64, in <module>
  File "D:/Pycharm/ValTest/parsetest.py", line 40, in convert_pdf_to_text
    text = retstr.getvalue()
  File "D:\Pycharm\ValTest\venv\lib\site-packages\pdfminer\pdfinterp.py", line 852, in process_page
    self.render_contents(page.resources, page.contents, ctm=ctm)
  File "D:\Pycharm\ValTest\venv\lib\site-packages\pdfminer\pdfinterp.py", line 864, in render_contents
    self.execute(list_value(streams))
  File "D:\Pycharm\ValTest\venv\lib\site-packages\pdfminer\pdfinterp.py", line 888, in execute
    func(*args)
  File "D:\Pycharm\ValTest\venv\lib\site-packages\pdfminer\pdfinterp.py", line 772, in do_TJ
    self.device.render_string(self.textstate, seq, self.ncs, self.graphicstate.copy())
  File "D:\Pycharm\ValTest\venv\lib\site-packages\pdfminer\pdfdevice.py", line 87, in render_string
    scaling, charspace, wordspace, rise, dxscale, ncs, graphicstate)
  File "D:\Pycharm\ValTest\venv\lib\site-packages\pdfminer\pdfdevice.py", line 100, in render_string_horizontal
    for cid in font.decode(obj):
  File "D:\Pycharm\ValTest\venv\lib\site-packages\pdfminer\pdffont.py", line 719, in decode
    return self.cmap.decode(bytes)
  File "D:\Pycharm\ValTest\venv\lib\site-packages\pdfminer\cmapdb.py", line 129, in decode
    return struct.unpack('>%dH' % n, code)
struct.error: unpack requires a buffer of 2 bytes

**Comment by pietermarsman** [comment]
It took me some time to understand what is going on here. The `IdentityCMap.decode()` function maps a byte string to a tuple of unsigned shorts (`H`). Each unsigned short needs two bytes, so a byte string of length 4 results in a tuple of 2 unsigned shorts (`>2H`). 

Another option in the format of `struct.unpack()` is the byte-order. Currently it is big-endian denoted by a `>`. You can also choose for little-endian denoted as `<` or native (`=`). Big-endian interprets the first bit of a byte as most significant, little-endian is the other way around ([wiki](https://en.wikipedia.org/wiki/Endianness#Examples)). 

The proposal in this PR is to change the endianness from big-endian to native. Native endianness lets the endianness depend on the host system.

I don't think that this will solve the issue in the error. The `struct.error` is the result of a mismatch between the code (i.e. the bytes) and the formatting. For example: 

```python
struct.unpack('>1H', b'\x00\x01')
(1,)
struct.unpack('>1H', b'\x00\x01\x02')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
struct.error: unpack requires a buffer of 2 bytes
```

Long story short, I don't think this PR solves any problem and can be closed. If I misinterpreted something or if I am just plainly wrong, please let me know.

**Comment by lucyk4t** [comment]
Thanks for the kind comment. 
I will check and try about it and this issue will be closed by me.
